internal/tiltfile/tiltfile_state.go

```diff
@@ -194,6 +194,11 @@ func (s *tiltfileState) builtins() starlark.StringDict {
 	return r
 }
 
+func (s *tiltfileState) warn(w string) {
+	s.logger.Infof("WARNING: %s\n", w)
+	s.warnings = append(s.warnings, w)
```

Comment:
maybe a comment on `s.warnings` to not manipulate that field directly, if it's not through the `warn` function?